client/app/components/DynamicComponent.jsx

```diff
@@ -1,50 +1,65 @@
-import { isFunction, isString } from "lodash";
-import React from "react";
-import PropTypes from "prop-types";
-
-const componentsRegistry = new Map();
-const activeInstances = new Set();
-
-export function registerComponent(name, component) {
-  if (isString(name) && name !== "") {
-    componentsRegistry.set(name, isFunction(component) ? component : null);
-    // Refresh active DynamicComponent instances which use this component
-    activeInstances.forEach(dynamicComponent => {
-      if (dynamicComponent.props.name === name) {
-        dynamicComponent.forceUpdate();
+import { isFunction, each } from "lodash";
+import React, { useState, useEffect } from "react";
+
+// Introduce new dynamic components here; use `null` for default representation (display children)
+const dynamicComponents = {
+  HomeExtra: null,
+  BeaconConsent: null,
+  BeaconConsentSetting: null,
+  UsersListExtra: null,
+  DataSourcesListExtra: null,
+  CreateDashboardDialogExtra: null,
+  HelpDrawerExtraContent: null,
+};
+
+// Overriding dynamic components is simple: just assign a value to one of the fields above, e.g.:
+//
+// import DynamicComponent from "@/components/DynamicComponent";
+// DynamicComponent.HomeExtra = CustomHomeExtraComponent;
+
+const updateListeners = new Set();
+
+// Patch components registry: convert each entry to getter + setter
+each(dynamicComponents, (Component, name, target) => {
```

Comment:
I'd consider a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) not to be doing it with each component + make it possible to add new DynamicComponents from other places.

Con: again, no IE 😅 (suppose it isn't a problem anymore, so I can bring the suggestion up anyways)